src/kafka/connection/kafka_connection.hh

```diff
@@ -127,6 +127,22 @@ public:
         auto request_future = send_request(request, api_version);
         auto response_future = request_future.then([this, api_version] (int32_t correlation_id) {
             return receive_response<RequestType>(correlation_id, api_version);
+        }).handle_exception([] (auto ep) {
+            try {
+                std::rethrow_exception(ep);
+            } catch (seastar::timed_out_error& e) {
+                typename RequestType::response_type response;
+                response._error_code = 7;
```

Comment:
Magic numbers will be gone when we merge Michał's error code support.

Reply:
good